group.go

```diff
@@ -134,6 +136,176 @@ func (client *Client) ListGroups() ([]Group, error) {
 	return q.Account.Groups.Query(client, &q, v)
 }
 
+func (g *Group) DescendantTeams(client *Client) ([]Team, error) {
+	var q struct {
+		Account struct {
+			Group struct {
+				DescendantTeams struct {
+					Nodes    []Team
+					PageInfo PageInfo
+				} `graphql:"descendantTeams(after: $after, first: $first)"`
+			} `graphql:"group(id: $group)"`
+		}
+	}
+	if g.Id == nil {
+		return nil, fmt.Errorf("failed to request 'DescendantTeams' for group because there is not a valid id '%s'", g.Id)
```

Comment:
I know i probably wrote this but reading it back upon review - it will be a confusing error for end users to read.

Do you have any ideas on how to make it clearer?
Maybe
```
fmt.Error("unable to get 'Teams' for group with missing id")
```

That or maybe we should just return an empty team array and no error.

```
output := []Team{}
if g.Id == nil {
    return output, nil
}
```

Reply:
Yeah, I like the idea of letting the user know that there's no Id on the group. Curious what scenario would cause that to happen, would it be that they passed in an invalid group?  If that's the case then I think we should let them know that in the error.

Reply:
@eochoa260 it could be as simple as just
```
group := opslevel.Group{}
team := group.DescendantTeams()
```
but it could also just be not error checking when
```
group, _ := client.GetGroup("group-does-not-exist")
team := group.DescendantTeams()
```